snippets/javascript.json

```diff
@@ -653,5 +653,868 @@
     "prefix": [
       "callout3DLargeProps"
     ]
+  },
+  "Add CSV Layer": {
+    "body": [
+    "CSVLayer({",
+    "\turl: \"${1:https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/2.5_week.csv}\",",
+    "\tcopyright: \"${2:USGS Earthquakes}\",",
+    "\tpopupTemplate: ${3:template}",
+    "});",
+    "map.add(csvLayer);"
+    ],
+    "description": "Add a CSVLayer with a URL",
+    "prefix": "csvLayer"
+  },
+  "Add CSV Layer from Blob": {
+    "body": [
+    "const blob = new Blob([${1:csv}], {",
+    "\ttype: \"plain/text\"",
+    "});",
+    "let url = URL.createObjectURL(blob);",
+    "",
+    "const layer = new CSVLayer({",
+    "\turl: ${2:url}",
+    "});",
+    "map.add(layer);"
+    ],
+    "description": "Create and add a CSVLayer from a Blob object",
+    "prefix": "csvLayerFromBlob"
+  },
+  "Add GeoJSONLayer with URL": {
+    "body": [
+    "GeoJSONLayer({",
+    "\turl: \"${1:https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_month.geojson}\",",
+    "\tcopyright: \"${2:USGS Earthquakes}\"",
+    "});",
+    "map.add(geojsonlayer);"
+    ],
+    "description": "Add GeoJSONLayer using URL",
+    "prefix": "geoJSONLayer"
+  },
+  "Create GeoRSSLayer with URL": {
+    "body": [
+    "GeoRSSLayer({",
+    "\turl: \"${1:https://arcgis.github.io/arcgis-samples-javascript/sample-data/layers-georss/sample-georss.xml}\"",
+    "});",
+    "map.add(geoRSSLayer);"
+    ],
+    "description": "Add GeoRSSLayer using URL",
+    "prefix": "addGeoRSSLayer"
+  },
+  "Add Graphics to GraphicsLayer": {
+    "body": [
+    "map.add(graphicsLayer);",  
+    "Graphic({",
+    "\tgeometry: ${1:point,polyline, polygon,}",
+    "\tsymbol: ${2:simpleMarkerSymbol,simpleLineSymbol,simpleFillSymbol,}",
+    "});",
+    "graphicsLayer.add(pointGraphic);"
+    ],
+    "description": "Create a graphic and add it to a GraphicsLayer",
+    "prefix": "graphicsLayer"
+  },
+  "Create FeatureLayer with PortalItem": {
+      "body": [
+        "FeatureLayer({",
+        "\tportalItem: {",
+        "\t\tid: \"${1:123f4410054b43d7a0bacc1533ceb8dc}\"",
+        "\t}",
+        "});",
+        "",
+        "table.load().then(function() {",
+        "\tif (table.isTable) {",
+        "\t\tmap.tables.add(table);",
+        "\t}",
+        "});"
+      ],
+      "description": "Create a FeatureLayer using a PortalItem ID",
+      "prefix": "featureLayerWithPortalItem"
+    },
+  "Create FeatureLayer with Client-Side Graphics": {
+      "body": [
+        "FeatureLayer({",
+        "\tfields: [",
+        "\t\t{",
+        "\t\t\tname: \"${1:ObjectID}\",",
+        "\t\t\talias: \"${2:ObjectID}\",",
+        "\t\t\ttype: \"${3:oid}\"",
+        "\t\t},",
+        "\t\t{",
+        "\t\t\tname: \"${4:type}\",",
+        "\t\t\talias: \"${5:Type}\",",
+        "\t\t\ttype: \"${6:string}\"",
+        "\t\t},",
+        "\t\t{",
+        "\t\t\tname: \"${7:place}\",",
```

Comment:
Probably we can remove this `place` field. It is redundant considering we have other "string field" already..